ipalib/errors.py

```diff
@@ -1697,6 +1697,15 @@ class CertificateFormatError(CertificateError):
     format = _('Certificate format error: %(error)s')
 
 
+class CertificateMappingError(CertificateError):
```

Comment:
What do you think of this version (https://github.com/freeipa/freeipa/pull/10/commits/6def0a5008c7da67374494c13f484503715223cb)? I subclassed SingleMatchExpected for the case of multiple matches for the helper, and merged the other two into CSRTemplateError with different descriptions.